### PR TITLE
Add RPL Hop-by-Hop option (RFC 6553)

### PR DESCRIPTION
**Pull request by thvdveld**
**Checklist:**

-   [x] If you are new to Scapy: I have checked [CONTRIBUTING.md](https://github.com/secdev/scapy/blob/master/CONTRIBUTING.md) (esp. section submitting-pull-requests)
-   [x] I squashed commits belonging together
-   [x] I added unit tests or explained why they are not relevant
-   [x] I executed the regression tests (using `cd test && ./run_tests` or `tox`)
-   [x] The PR is complete

This adds support for the RPL Option, defined in [RFC 6553](https://www.rfc-editor.org/rfc/rfc6553.html), which is used in the Hop-by-Hop options.

### CODE REVIEW

**Review by guedou — COMMENTED** [review]
This looks good to me. Could you explain us what the parsing of the sub-TLV should be?

**Comment by thvdveld** [comment]
The RFC says that there are no sub-TLV's defined in that RFC.

> The action taken by using the RPL Option and the potential set of
   sub-TLVs carried within the RPL Option MUST be specified by the RFC
   of the protocol that uses that option.  No sub-TLVs are defined in
   this document.  A RPL device MUST skip over any unrecognized sub-TLVs
   and attempt to process any additional sub-TLVs that may appear after.

I hope this answers your question.

File: scapy/layers/inet6.py [hunk 1]
```diff
@@ -60,6 +60,12 @@
     XIntField,
     XShortField,
 )
+from scapy.fields import BitEnumField, BitField, ByteEnumField, ByteField, \
```
Comment (gpotter2) [inline scapy/layers/inet6.py:63]: Sorry but this is duplicated. Looks like a rebase mistake.

Comment (thvdveld) [reply scapy/layers/inet6.py]: Sorry for that!

File: scapy/layers/inet6.py [hunk 2]
```diff
@@ -57,9 +57,11 @@
     StrLenField,
     X3BytesField,
     XBitField,
+    XByteField,
     XIntField,
     XShortField,
 )
+
```
Comment (p-l-) [inline scapy/layers/inet6.py:64]: Hi,
I know it looks like nothing but we try to keep patches as tiny as possible (to keep our history as clean as possible and avoid unneeded issues for people maintaining draft PRs), and remove unnecessary changes
```suggestion
```

Comment (thvdveld) [reply scapy/layers/inet6.py]: Can I find formatting guidlines somewhere? If I run a formatter on the file, the whole file changes.

**Comment by guedou** [comment]
@thvdveld the linting issues will appear in the `Checks` tab. Locally, you can install https://tox.wiki/en/4.11.4/ and launch `tox flake8` to reproduce the CI behavior.

**Comment by thvdveld** [comment]
@guedou Thank you!

**Review by gpotter2 — APPROVED** [review]

**Comment by gpotter2** [comment]
Very sorry for the delay !
